component/azstorage/config.go

```diff
@@ -390,12 +390,26 @@ func ParseAndValidateConfig(az *AzStorage, opt AzStorageOptions) error {
 		az.stConfig.authConfig.SASKey = sanitizeSASKey(opt.SaSKey)
 	case EAuthType.MSI():
 		az.stConfig.authConfig.AuthMode = EAuthType.MSI()
-		if opt.ApplicationID == "" && opt.ResourceID == "" {
+		if opt.ApplicationID == "" && opt.ResourceID == "" && opt.ObjectID == "" {
 			//lint:ignore ST1005 ignore
 			return errors.New("Application ID and Resource ID not provided")
 		}
+		v := make(map[string]bool, 3)
+		if opt.ClientID != "" {
+			v[opt.ClientID] = true
+		}
+		if opt.ObjectID != "" {
+			v[opt.ObjectID] = true
+		}
```

Comment:
What is use of these flags?

Reply:
this is just a validation step to confirm only one of them is provided